diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt

```diff
@@ -189,38 +198,55 @@ class IdentifierNaming(configRules: List<RulesConfig>) : DiktatRule(
                         }
                     }
                 } else if (variableName.text != "_" && !variableName.text.isLowerCamelCase()) {
+                    var isCorrectBackingField = false
+                    if (variableName.text.commonPrefixWith("_").length == 1 && variableName.text.drop(1).isLowerCamelCase()) {
+                        val matchingNode = propertyNodes.find { propertyNode ->
+                            val nodeType = node.getFirstChildWithType(TYPE_REFERENCE)
+                            val propertyType = propertyNode.getFirstChildWithType(TYPE_REFERENCE)
+                            propertyNode.getFirstChildWithType(IDENTIFIER)?.text == variableName.text.drop(1) &&
+                                    (propertyType?.text == nodeType?.text ||
+                                            propertyType?.getFirstChildWithType(USER_TYPE)?.text ==
+                                                    nodeType?.getFirstChildWithType(NULLABLE_TYPE)?.getFirstChildWithType(USER_TYPE)?.text) &&
+                                    node.getFirstChildWithType(MODIFIER_LIST)?.getFirstChildWithType(PRIVATE_KEYWORD) != null &&
+                                    node.getFirstChildWithType(PROPERTY_ACCESSOR) == null &&
+                                    propertyNode.getFirstChildWithType(PROPERTY_ACCESSOR) != null
+                        }
+                        matchingNode.let { isCorrectBackingField = true }
```

Comment:
probably you wanted to check on null here
```suggestion
                        matchingNode?.let { isCorrectBackingField = true }
```

Reply:
Done